command/head.go

```diff
@@ -0,0 +1,300 @@
+			&cli.BoolFlag{
+				Name:    "etag",
+				Aliases: []string{"e"},
+				Usage:   "show entity tag (ETag) in the output",
+			},
+			&cli.BoolFlag{
+				Name:    "humanize",
+				Aliases: []string{"H"},
+				Usage:   "human-readable output for object sizes",
+			},
+			&cli.BoolFlag{
+				Name:    "storage-class",
+				Aliases: []string{"s"},
+				Usage:   "display full name of the object class",
+				Value:   true,
+			},
+
+			&cli.BoolFlag{
+				Name:  "all-versions",
+				Usage: "list all versions of object(s)",
+			},
+			&cli.BoolFlag{
+				Name:  "show-fullpath",
+				Usage: "shows only the fullpath names of the object(s)",
+			},
+		},
+
+		Before: func(c *cli.Context) error {
+			err := validateHEADCommand(c)
+			if err != nil {
+				printError(commandFromContext(c), c.Command.Name, err)
+			}
+			return err
+		},
+		Action: func(c *cli.Context) (err error) {
+
+			//print the command name
+			//fmt.Println(c.Command.FullName())
+
+			defer stat.Collect(c.Command.FullName(), &err)()
+
+			//fmt.Println("head command")
+
+			op := c.Command.Name
+			//fmt.Println("op: ", op)
+			fullCommand := commandFromContext(c)
+
+			//print the command name
+			//fmt.Println("fullCommand: ", fullCommand)
+
+			src, err := url.New(c.Args().Get(0), url.WithVersion(c.String("version-id")),
+				url.WithRaw(c.Bool("raw")))
+			if err != nil {
+				printError(fullCommand, op, err)
+				return err
+			}
+
+			return Head{
+				src:         src,
+				op:          op,
+				fullCommand: fullCommand,
+				//flag
+				showEtag:         c.Bool("etag"),
+				humanize:         c.Bool("humanize"),
+				showStorageClass: c.Bool("storage-class"),
+				showFullPath:     c.Bool("show-fullpath"),
+
+				storageOpts: NewStorageOpts(c),
+			}.Run(c.Context)
+
+		},
+	}
+	cmd.BashComplete = getBashCompleteFn(cmd, true, false)
+	return cmd
+}
+
+type Head struct {
+	src         *url.URL
+	op          string
+	fullCommand string
+
+	showEtag         bool
+	humanize         bool
+	showStorageClass bool
+	showFullPath     bool
+
+	storageOpts storage.Options
+}
+
+func (h Head) Run(ctx context.Context) error {
+	h.src.SetRelative(h.src)
+	client, err := storage.NewRemoteClient(ctx, h.src, h.storageOpts)
+	if err != nil {
+		printError(h.fullCommand, h.op, err)
+		return err
+	}
+
+	if h.src.IsBucket() {
+		bucket, err := client.HeadBucket(ctx, h.src)
+		if err != nil {
+			printError(h.fullCommand, h.op, err)
+			return err
+		}
+
+		msg := HeadBucketMessage{
+			Bucket: bucket,
+		}
+
+		log.Info(msg)
+
+		return nil
+
+	}
+
+	object, metadata, err := client.HeadObject(ctx, h.src)
+
+	//print the metadata all fields
+
+	//fmt.Println("metadata: ", metadata)
+
+	if err != nil {
+		printError(h.fullCommand, h.op, err)
+		return err
+	}
+
+	msg := HeadObjectMessage{
+		Object:           object,
+		showEtag:         h.showEtag,
+		showHumanized:    h.humanize,
+		showStorageClass: h.showStorageClass,
+		showFullPath:     h.showFullPath,
+		Metadata:         metadata,
```

Comment:
unused. output for objects is basically the same as `ls`. 

if metadata is not to be used, you can remove it. if our goal is to keep same format we can even go further and do this;
```go
type HeadObjectMessage ListMessage

func (h HeadObjectMessage) JSON() string {
	return ListMessage(h).JSON()
}

func (h HeadObjectMessage) String() string {
	return ListMessage(h).String()
}

```

until they start to differentiate.

Reply:
### new `JSON()` function for `HeadObjectMessage`

```go
func (m HeadObjectMessage) JSON() string {
	j := struct {
		storage.Object
		Metadata map[string]string `json:"metadata"`
	}{
		Object:   *m.Object,
		Metadata: m.Metadata,
	}

	return strutil.JSON(j)
}
```

#### Output json

```json
{
    "key": "s3://my-local-bucket/sample-file.txt",
    "etag": "3c1f41bc58c38a0137d9f5dd712c40af",
    "last_modified": "2024-07-05T11:34:38Z",
    "type": "file",
    "size": 25,
    "storage_class": "STANDART",
    "metadata": {
        "x-amz-meta-custom-header": "custom-value"
    }
}
```